docs/DafnyRef/Topics.md

```diff
@@ -10,6 +10,31 @@ TO BE WRITTEN
 
 TO BE WRITTEN
 
+### 23.2.1. Subset types
+
+[Subset types](#sec-subset-types) 
+Most of the time, subset types are carrying extra constraints so, unless it is trivial to infer them, only their base type is inferred, and the verifier checks the constraints.
```

Comment:
My assumption is that if a compiler can not infer a type for me, then I have to explicitly provide the type, but I believe that's not what you're trying to say here, is it?